### PR TITLE
Configure mapping for Saga acceptance tests

### PR DESCRIPTION
**Pull request by timbussmann**
@danielmarbach @DavidBoike 
related to #324

We decided that Sagas in acceptance tests should be configured in a realistic way. Therefore I configured the message mappings for test Sagas with empty `ConfigureHowToFindSaga(...)` methods.

### CODE REVIEW

File: src/NServiceBus.AcceptanceTests/Sagas/When_doing_request_response_between_sagas.cs [hunk 1]
```diff
@@ -125,7 +125,15 @@ void SendReply()
...
 
-        public class InitiateRequestingSaga : ICommand { }
+        public class InitiateRequestingSaga : ICommand
+        {
+            public InitiateRequestingSaga()
+            {
+                Id = Guid.NewGuid();
+            }
+
+            public Guid Id { get; private set; }
```
Comment (DavidBoike) [inline src/NServiceBus.AcceptanceTests/Sagas/When_doing_request_response_between_sagas.cs]: Not sure it's good to have a message property as a private setter. Couldn't this mess with deserialization and result in mismatched Guids?

File: src/NServiceBus.AcceptanceTests/Sagas/When_sagas_cant_be_found.cs [hunk 2]
```diff
@@ -211,6 +227,12 @@ public void Handle(object message)
         [Serializable]
         public class StartSaga : ICommand
         {
+            public StartSaga()
+            {
+                Id = Guid.NewGuid();
+            }
+
+            public Guid Id { get; private set; }
```
Comment (DavidBoike) [inline src/NServiceBus.AcceptanceTests/Sagas/When_sagas_cant_be_found.cs]: Private setter again here.

File: src/NServiceBus.AcceptanceTests/Sagas/When_sagas_cant_be_found.cs [hunk 3]
```diff
@@ -221,6 +243,12 @@ public class FinishMessage : ICommand
         [Serializable]
         public class MessageToSaga : ICommand
         {
+            public MessageToSaga()
+            {
+                Id = Guid.NewGuid();
+            }
+
+            public Guid Id { get; private set; }
```
Comment (DavidBoike) [inline src/NServiceBus.AcceptanceTests/Sagas/When_sagas_cant_be_found.cs]: Another private setter here.

**Comment by DavidBoike** [comment]
Some of the code raises an important realization. If you do a custom saga finder, it doesn't make sense to have anything in ConfigureHowToFindSaga. Yet, the definition of the Saga is no different. Something to keep in mind should we ever entertain the idea about asserting anything about what happens inside the ConfigureHowToFindSaga.

**Comment by timbussmann** [comment]
fixed the private setters.

I think it's also tedious to have to check each Message for it's usage to determine the correct mapping to the Saga. That's probably obvious for timeout messages but not self-evident when using responses.

**Pull request merged**